bin/git-cp

```diff
@@ -38,6 +38,18 @@ else
         exit 40
     fi
 
+    if [[ "$DESTINATION_FILENAME" == */ ]]; then
+      echo 1>&2 "$DESTINATION_FILENAME is not a file path."
+      exit 80
+    fi
+    DESTINATION_DIR="${DESTINATION_FILENAME%/*}"
```

Comment:
What if the DESTINATION_FILENAME doesn't contain the directory, like `./foo`?

Reply:
Good question! Although the new code still works for `./foo` (because `DESTINATION_DIR` will simply be `./` and creating it is a harmless no-op), another case that I tested did lead to incorrect behavior which is when the `DESTINATION_FILENAME ` is just a filename. The previous revision would create a folder of that name, and make a copy of the source file into that folder (rather than copy to a file of that name).

I've now switched the condition to be checking whether `DESTINATION_FILENAME` contains any slash characters - if not the folder creation is skipped.